### ISSUE TITLE
Ignoring variables with underscore. 

### ISSUE THREAD

**Issue by AndrewVasilevskii** [heftapp/graphql_codegen#5]
In case if model contains variable with underscore ex. `_id`, it's ignored and not used.

**Comment by mvarendorff** [comment]
I believe the reason is that underscore prefixed variables are private in Dart so it doesn't make sense to generate those properties (unless you change the prefix to $ for example). 

The context determines publicProperties by that exact condition:

https://github.com/heftapp/graphql_codegen/blob/a8ab0e883b4c04360377fdd09e7bb7957026c9fb/graphql_codegen/lib/src/context.dart#L476-L477

And only considers public properties when generating a class:

https://github.com/heftapp/graphql_codegen/blob/a8ab0e883b4c04360377fdd09e7bb7957026c9fb/graphql_codegen/lib/src/printer/printer.dart#L326-L328

I created #3 a couple of days ago to allow the __typename property to generate, the diff should point to both places that will require a change to generate properties with leading underscores to function properly.

**Comment by budde377** [comment]
Sorry for the delayed response. The recommended way to handle public fields is to alias them, i.e.:

```graphql
query Q { 
  typename: __typename
}
```
I am open to making this behavior configurable.

**Comment by alexkharech** [comment]
Prisma uses an underscore scheme.
```gql
_count
_min
_max
_avg
_sum
```
if the type has one of the following fields: min, max, avg, etc...,
repeating fields are obtained

**Comment by budde377** [comment]
@alexkharech, I am not sure what you mean. Are you not able to alias these fields?

**Comment by alexkharech** [comment]
Error generating `scheme.graphql.g.dart` file.
A good solution would be to replace the underscores with $ like Artemis does. Don't cut them out completely.

**Comment by budde377** [comment]
In this case your query wouldn't be valid. Here I'd alias it to something different:

‘‘‘
min
secretMin: _min

‘‘‘

**Comment by budde377** [comment]
I've changed the naming strategy, so this should now be supported out of the box.

**Comment by erdzan12** [comment]
is it only included in the alpha [0.7.0-beta.3](https://pub.dev/packages/graphql_codegen/versions/0.7.0-beta.3)?

**Comment by budde377** [comment]
> is it only included in the alpha [0.7.0-beta.3](https://pub.dev/packages/graphql_codegen/versions/0.7.0-beta.3)?

Yes!
